### PR TITLE
chore: add monorepo tools

### PR DESCRIPTION
**Pull request by forehalo**

### CODE REVIEW

**Comment by forehalo** [comment]
* **#9196** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/9196?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/9196?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `canary`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/toeverything?ref=merge-queue-instructions-in-stack-comment&prId=9196).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: packages/frontend/i18n/src/i18n.gen.ts [hunk 1]
```diff
@@ -0,0 +1,7180 @@
+// @ts-nocheck
```
Comment (darkskygit) [inline packages/frontend/i18n/src/i18n.gen.ts:1]: seems should be ignore

Comment (forehalo) [reply packages/frontend/i18n/src/i18n.gen.ts:1]: the i18n codegen will not be required on installing

**Comment by forehalo** [comment]
### Merge activity

* **Dec 23, 7:57 AM EST**: [Graphite](https://app.graphite.dev/github/pr/toeverything/AFFiNE/9196) disabled "merge when ready" on this PR due to: a merge conflict with the target branch; resolve the conflict and try again..

**Review by EYHN — APPROVED** [review]

**Review by pengx17 — COMMENTED** [review]
Can you please list the improvements of the new monorepo tool?

**Comment by forehalo** [comment]
> Can you please list the improvements of the new monorepo tool?

there is a simple readme at `tools/cli/README.md` for it and I will add more detail once it become more stable. this PR is going to be bigger

File: tools/utils/src/process.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+      logger.error('Finished with non-zero exit code.');
+    }
+
+    drain(code, signal);
+  });
+
+  return childProcess;
+}
+
+export function execAsync(
```
Comment (darkskygit) [inline tools/utils/src/process.ts:67]: seems equal to util.promisify(spawn)

Comment (forehalo) [reply tools/utils/src/process.ts:67]: it bring more features than that, a tag is passed to separate different process's stdout

**Pull request merged**